### PR TITLE
Use experimental ASM 7 mode

### PR DESCRIPTION
**Pull request by oehme**
For full Java 11 compatibility

### CODE REVIEW

File: subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/GroovyCompileTransformingClassLoader.java [hunk 1]
```diff
@@ -94,21 +99,21 @@ public AnnotationVisitor visitAnnotation(String desc, boolean visible) {
...
+                    return new AnnotationVisitor(AsmConstants.ASM_LEVEL){
                         @Override
                         public void visit(String name, Object value) {
                             Type type = (Type) value;
                             names.add(type.getClassName());
                         }
                     };
                 } else if (name.equals("value")) {
-                    return new AnnotationVisitor(Opcodes.ASM6) {
+                    return new AnnotationVisitor(Opcodes.ASM7_EXPERIMENTAL) {
```
Comment (eskatos) [inline subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/GroovyCompileTransformingClassLoader.java]: :x: `AsmConstants.ASM_LEVEL`

**Comment by oehme** [comment]
Unfortunately this is still not enough, because cglib uses ASM6, so a lot of our Spock tests fail on Java 11. We won't be able to announce compatibility in this state. It's a bit upsetting that such major changes come so late, when it's supposed to be in "rampdown".

**Comment by marcphilipp** [comment]
On the upside at least we found out about this issue before announcing support for Java 11. Sounds like it will have to wait until 5.0, doesn't it?

**Comment by oehme** [comment]
Yes, I'll ping the Spock team about a fix.

**Comment by ijuma** [comment]
Out of curiosity, gradle is not affected by https://github.com/cglib/cglib/pull/130 ?

**Comment by marcphilipp** [comment]
@oehme I changed the milestone to 5.0 RC1.

File: subprojects/internal-testing/internal-testing.gradle.kts [hunk 2]
```diff
@@ -27,6 +27,7 @@ dependencies {
     compile(library("groovy"))
 
     compile(project(":baseServices"))
+    compile(project(":core"))
```
Comment (lptr) [inline subprojects/internal-testing/internal-testing.gradle.kts]: Can we avoid this by putting `AsmConstants` in `base-services`?

Comment (oehme) [reply subprojects/internal-testing/internal-testing.gradle.kts]: yep

**Comment by ijuma** [comment]
As an aside, ASM 6.2.1 has been released https://mail.ow2.org/wws/arc/asm/2018-08/msg00000.html

**Pull request merged**